crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -718,8 +723,19 @@ impl Timeline {
             return Ok(false);
         }
 
-        trace!("sending receipt");
-        self.room().send_single_receipt(receipt_type, thread, event_id).await?;
+        // Since neither `ReceiptType` nor `ReceiptThread` implement Hash/Ord, we
+        // "flatten" them to a string, which we use to uniquely identify the
+        // query.
+        let request_key = format!("{}|{}", receipt_type, thread.as_str().unwrap_or("<unthreaded>"));
```

Comment:
Hmm, why not put the deduplication into the `send_single_receipt()` method?

Reply:
... Very good point :grin:

Reply:
That was a great idea, and I don't even need to make the deduplicating handler public now. Thanks!